src/lib/child_processes/child_processes.ml

```diff
@@ -111,49 +111,60 @@ let maybe_kill_and_unlock : string -> Filename.t -> Logger.t -> unit Deferred.t
     =
  fun name lockpath logger ->
   let open Deferred.Let_syntax in
+  let try_cleanup_lock_file ~pid_metadata () =
+    match%bind Sys.file_exists lockpath with
+    | `Yes | `Unknown -> (
+        match%bind try_with ~here:[%here] (fun () -> Sys.remove lockpath) with
+        | Ok () ->
+            Deferred.unit
+        | Error exn ->
+            [%log warn]
+              !"Couldn't delete lock file for %s (pid $childPid). If another \
+                Mina daemon was already running it may have cleaned it up for \
+                us. ($exn)"
+              name
+              ~metadata:
+                [ ("childPid", pid_metadata)
+                ; ("exn", `String (Exn.to_string exn))
+                ] ;
+            Deferred.unit )
+    | `No ->
+        Deferred.unit
+  in
   match%bind Sys.file_exists lockpath with
-  | `Yes -> (
+  | `Yes ->
       let%bind pid_str = Reader.file_contents lockpath in
-      let pid = Pid.of_string pid_str in
-      [%log debug] "Found PID file for %s %s with contents %s" name lockpath
-        pid_str ;
-      let%bind () =
-        match Signal.send Signal.term (`Pid pid) with
-        | `No_such_process ->
-            [%log debug] "Couldn't kill %s with PID %s, does not exist" name
-              pid_str ;
-            Deferred.unit
-        | `Ok -> (
-            [%log debug] "Successfully sent TERM signal to %s (%s)" name pid_str ;
-            let%bind () = after (Time.Span.of_sec 0.5) in
-            match Signal.send Signal.kill (`Pid pid) with
-            | `No_such_process ->
-                Deferred.unit
-            | `Ok ->
-                [%log error]
-                  "helper process %s (%s) didn't die after being sent TERM, \
-                   KILLed it"
-                  name pid_str ;
-                Deferred.unit )
-      in
-      match%bind Sys.file_exists lockpath with
-      | `Yes | `Unknown -> (
-          match%bind try_with ~here:[%here] (fun () -> Sys.remove lockpath) with
-          | Ok () ->
+      if String.is_empty pid_str then (
```

Comment:
I would try to be more generic than this, like if the file is corrupted, i.e. it does not contain a valid PID, we cleanup the lockfile.